modules/dnn/src/layers/max_unpooling_layer.cpp

```diff
@@ -181,6 +183,44 @@ class MaxUnpoolLayerImpl CV_FINAL : public MaxUnpoolLayer
 #endif  // HAVE_HALIDE
         return Ptr<BackendNode>();
     }
+
+#ifdef HAVE_DNN_NGRAPH
+    virtual Ptr<BackendNode> initNgraph(const std::vector<Ptr<BackendWrapper> >& inputs,
+                                        const std::vector<Ptr<BackendNode> >& nodes) CV_OVERRIDE
+    {
+        auto features = nodes[0].dynamicCast<InfEngineNgraphNode>()->node;
+        auto indices = nodes[1].dynamicCast<InfEngineNgraphNode>()->node;
+
+        auto inpShape = features->get_shape();
+        std::vector<int32_t> outShape(inpShape.begin(), inpShape.end());
+        outShape[2] = (outShape[2] - 1) * poolStride.height + poolKernel.height - 2 * poolPad.height;
```

Comment:
I propose to add assert to check size of `outShape` before access

Reply:
Replaced to manual call of `getMemoryShapes`